src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Contracts/QueryExecuteRequest.cs

```diff
@@ -32,6 +32,12 @@ public class QueryExecuteResult
         /// Connection error messages. Optional, can be set to null to indicate no errors
         /// </summary>
         public string Messages { get; set; }
```

Comment:
In an ideal world we'd use one of 2 patterns:
1. Each message has a MessageType (Error, Warn, Info). 
2. We have 2 streams, 1 for error messages, the other for InfoMessages.

This would be less restrictive than the current approach. It's not vital and your solution requires minimal code change on the mssql extension side, but it's something to think about and see if it's worth it